O2/ML/compute_efficiency_df.py

```diff
@@ -0,0 +1,193 @@
+"""
+Script for BDT efficiency calculation
+"""
+
+import argparse
+import numpy as np
+#import itertools
+import pandas as pd
+import yaml
+import matplotlib.pyplot as plt
+
+def config_bins(pt_bins):
+    """
+    Helper method to configure the bins to plot efficiency vs pT
+
+    --------------------------------
+    Parameters
+    - pt_bins: array containing the pT values used for binning
+    --------------------------------
+    Outputs
+    - bins_width: array with the width of each bin
+    - bins_mean: array with the mean of each bin
+    """
+    pt_mins = pt_bins[:-1]
+    pt_maxs = pt_bins[1:]
+    bins_width = []
+    bins_mean = []
+    for ipt, (pt_min, pt_max) in enumerate(zip(pt_mins, pt_maxs)):
+        bins_width.append(0.5*(pt_max - pt_min))
+        bins_mean.append(0.5*(pt_min + pt_max))
+    return bins_width, bins_mean
+
+def find_cut_indices(thresholds, BDT_cuts):
+    """
+    Helper method to find the index corresponding to chosen BDT cut in the 3D array eff
+
+    --------------------------------
+    Parameters
+    - thresholds: array of BDT cuts to browse
+    - BDT_cuts: array of chosen BDT output values for each class
+    --------------------------------
+    Outputs
+    - effifiency: 3D array efficiency[pT index][class][efficiency value]
+    """
+    idx_list = []
+    for cut in BDT_cuts:
+        idx, = np.where(thresholds == cut)
+        idx_list.append(idx[0])
+    return idx_list
+
+def compute_eff_vs_pt(eff, cuts_indices):
+    """
+    Helper method to compute efficiency vs pT
+    
+    --------------------------------
+    Parameters
+    - eff: output of compute_efficiency() method
+    - cuts_indices: output of find_cuts_indices() method
+    --------------------------------
+    Outputs
+    - effifiency: 3D array efficiency[pT index][class][efficiency value]
+    """
+    eff_vs_pt = []
+    for clas in range(3):
+        eff_vs_pt.append([])
+        for ipt in range(len(eff)):
+            eff_vs_pt[clas].append(eff[ipt][clas][cuts_indices[clas]])
```

Comment:
```suggestion
    Also this can be a dictionary
    eff_vs_pt = {}
    for iclass, clas in enumerate(cuts_indices):
        eff_vs_pt[class] = []
        for ipt, _ in enumerate(eff):
            eff_vs_pt[clas].append(eff[ipt][iclass][cuts_indices[clas]])
```